### PR TITLE
#69680 - Authorize retry interceptor

### PR DESCRIPTION
**Pull request by TomasWahlgren**

### CODE REVIEW

File: services/authorize/client.go [hunk 1]
```diff
@@ -2,13 +2,15 @@ package authorize
 
 import (
 	"context"
+	"google.golang.org/grpc/codes"
```
Comment (j-sv) [inline services/authorize/client.go:5]: Move this import together with the other 3pp imports.

File: services/authorize/client.go [hunk 2]
```diff
@@ -207,6 +205,20 @@ func (c *client) DialUsingCredentialsWithContext(ctx context.Context, sess *sess
...
+	retryIC := grpc_retry.UnaryClientInterceptor(
+		grpc_retry.WithBackoff(grpc_retry.BackoffLinear(100*time.Millisecond)),
+		grpc_retry.WithCodes(codes.Unavailable, codes.ResourceExhausted, codes.Aborted),
+	)
+
+	reconnectIC := reconnect.UnaryInterceptor(
+		reconnect.WithNewConnection(newClientConn),
+	)
+
+	dialOptsReconnectRetry := grpc.WithChainUnaryInterceptor(reconnectIC, retryIC) // first one is outer, being called last
```
Comment (j-sv) [inline services/authorize/client.go:218]: No need to create a variable, return it directly.

File: services/authorize/client.go [hunk 3]
```diff
@@ -207,6 +205,20 @@ func (c *client) DialUsingCredentialsWithContext(ctx context.Context, sess *sess
 	return err
 }
 
+func reconnectRetryInterceptor(newClientConn reconnect.NewConnectionFunc) grpc.DialOption {
+	retryIC := grpc_retry.UnaryClientInterceptor(
+		grpc_retry.WithBackoff(grpc_retry.BackoffLinear(100*time.Millisecond)),
```
Comment (j-sv) [inline services/authorize/client.go:210]: I think this should be configurable. Even which type of backoff is used.

Comment (TomasWahlgren) [reply services/authorize/client.go:210]: That can be a separate ticket for improvements

File: services/authorize/client.go (hunk 2 above)
Comment (TomasWahlgren) [reply services/authorize/client.go:218]: Code gets so messy, rather not

File: services/authorize/client.go (hunk 1 above)
Comment (TomasWahlgren) [reply services/authorize/client.go:5]: fixing

**Comment by TomasWahlgren** [comment]
We'll add the optionality to the follow up PR ticket

**Pull request merged**